### PR TITLE
Add DPGD

### PR DESCRIPTION
**Pull request by EnLAI111**
I tried to add Dual proximal gradient descent, however, it can't converge. It is waiting for me to figure out why.

### CODE REVIEW

File: solvers/DPGD.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    """Dual Projected gradient descent for analysis formulation."""
+    name = 'Dual PGD analysis'
+
+    stopping_criterion = SufficientDescentCriterion(
+        patience=3, strategy="callback"
+    )
+
+    # any parameter defined here is accessible as a class attribute
+    parameters = {'alpha': [1.],
+                  'use_acceleration': [False, True]}
```
Comment (tomMoral) [inline solvers/DPGD.py]: This is not used anywhere?

File: solvers/DPGD.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+        while callback(u):
+            v_tmp, _ = cg(AtA, (Aty + div(vh, vv)).flatten(),
+                          x0=v_tmp.flatten(), tol=tol_cg)
+            v_tmp = v_tmp.reshape((n, m))
+            vh, vv = proj(vh + stepsize * grad(v_tmp)[0],
+                          vv + stepsize * grad(v_tmp)[1],
+                          self.reg)
+            u, _ = cg(AtA, (Aty + div(vh, vv)).flatten(),
+                      x0=u.flatten(), tol=tol_cg)
```
Comment (tomMoral) [inline solvers/DPGD.py]: This looks weird that it is the same as `v_tmp` at the next iteration, up to the starting point?
You should either consider that the next `v_tmp` is equal to `u` or their is a bug?

File: solvers/DPGD.py (hunk 1 above)
Comment (EnLAI111) [reply solvers/DPGD.py]: I haven't added codes about acceleration, because without acceleration the solvers can't converge yet.

**Comment by svaiter** [comment]
Don't you need A to be an orthogonal projection to ensure convergence? I don't think the algorithm is valid in 2D for deconvolution.

**Comment by EnLAI111** [comment]
Now this solver only works in the case of denoising, that is to say, when $A$ is identity.
I have found a paper https://epubs.siam.org/doi/pdf/10.1137/S1064827596299767.
In the page 1971 and 1972, there are pseudo-codes.
Moreover, I tried to solve the equation (4.7) between primal variable and dual variable in page 1970 by ```scipy.optimize.minimize``` with mode ```BFGS```, which is $A^{\top}A \boldsymbol{u} =  \triangledown \cdot \boldsymbol{v} + A^{\top} \boldsymbol{y}$ with our symbols.
However, the ```scipy.optimize.minimize``` always fails.

**Pull request merged**